soc/arm/st_stm32/stm32wl/soc.c

```diff
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 2020 STMicroelectronics
+ * Modifications copyright (C) 2023 SteadConnect
```

Comment:
Is this a thing? Copyright headers don't normally get edits for every individual modifications, especially bug fixes like this one. I could understand major code reworks but I'd suggest dropping this in this case, imagine if everyone adds their line to every file, we'd have more copyrights than code :-)

Reply:
I absolutely agree that it seems like overkill. But it seems to be required by section 4 of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0): "You must cause any modified files to carry prominent notices stating that You changed the files". This is repeated on the [WikiPedia page](https://en.wikipedia.org/wiki/Apache_License) for the license. It doesn't include language like "significant change" to provide an opt-out. 

See [here](https://softwareengineering.stackexchange.com/questions/220068/file-with-apache-2-0-and-my-modifications), [here](https://opensource.stackexchange.com/questions/8096/how-can-i-correctly-apply-the-apache-2-0-licence-to-contributed-code-from-an-exi), and [here](https://stackoverflow.com/questions/37781587/what-should-i-note-when-i-want-to-change-apache-license-2-0-code) for a few times where this has come up.

Reply:
Interesting, did not realize it was spelled out like that in the licence, but I see what you mean:

https://github.com/zephyrproject-rtos/zephyr/blob/89bcb874db5271b4d6c7abb28e856275379ee875/LICENSE#L97-L98

@nashif @kartben do you we have a guidance for this? Wondering if it's been discussed before.

Reply:
I mean, the Apache code itself does not have any of these: https://github.com/apache/httpd/tree/trunk/server

Reply:
Thanks, it's a fair argument though and should probably be clarified, I'll keep asking around, thanks for pointing it out.

Reply:
I think that the mentioned line only concerns [redistribution](https://github.com/zephyrproject-rtos/zephyr/blob/89bcb874db5271b4d6c7abb28e856275379ee875/LICENSE#L89), which is not the current situation.
Current case is `5. Submission of Contributions`, , which isn't quite explicit about copyright notice. https://github.com/zephyrproject-rtos/zephyr/blob/89bcb874db5271b4d6c7abb28e856275379ee875/LICENSE#L130-L136

Reply:
BTW can you reapprove @erwango?